### PR TITLE
[Keycloak] Make parsers configurable to support multiline logs

### PR DESCRIPTION
**Pull request by chemamartinez**
## Proposed commit message

In order to process multiline logs, this pull request makes the parsers configurable for the Keycloak integration.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] Added multiline cases to pipeline tests

## Screenshots
image

### CODE REVIEW

File: packages/keycloak/data_stream/log/agent/stream/filestream.yml.hbs [hunk 1]
```diff
@@ -17,9 +17,13 @@ processors:
 {{#if processors}}
 {{processors}}
 {{/if}}
+{{#if parsers}}
+parsers:
+{{parsers}}
+{{/if}}
```
Comment (efd6) [inline packages/keycloak/data_stream/log/agent/stream/filestream.yml.hbs:23]: This is resulting in the processors below this being added into the `parsers:` group. Suggest this instead:
```
{{#if parsers}}
parsers:
{{parsers}}
{{/if}}
processors:
{{#if processors}}
{{processors}}
{{/if}}
- add_locale: ~
- add_fields:
    target: _tmp
    fields:
      tz_offset: "{{tz_offset}}"
      only_user_events: {{only_user_events}}
```

File: packages/keycloak/data_stream/log/_dev/test/pipeline/test-log.log [hunk 2]
```diff
@@ -19,4 +19,11 @@
...
 2021-10-22 22:46:14,913 DEBUG [org.keycloak.events] (default task-1) operationType=DELETE, realmId=test, clientId=7bcaf1cb-820a-40f1-91dd-75ced03ef03b, userId=ce637d23-b89c-4fca-9088-1aea1d053e19, ipAddress=10.2.2.156, resourceType=GROUP, resourcePath=groups/d043d5af-6100-483a-9c41-b1a30d5149f7
-2021-10-22 23:05:03,371 DEBUG [org.keycloak.events] (default task-8) operationType=CREATE, realmId=test, clientId=7bcaf1cb-820a-40f1-91dd-75ced03ef03b, userId=ce637d23-b89c-4fca-9088-1aea1d053e19, ipAddress=10.2.2.156, resourceType=GROUP, resourcePath=groups/a57cd49f-fdfd-4d25-9fd2-2a46de44a9e6/children
\ No newline at end of file
+2021-10-22 23:05:03,371 DEBUG [org.keycloak.events] (default task-8) operationType=CREATE, realmId=test, clientId=7bcaf1cb-820a-40f1-91dd-75ced03ef03b, userId=ce637d23-b89c-4fca-9088-1aea1d053e19, ipAddress=10.2.2.156, resourceType=GROUP, resourcePath=groups/a57cd49f-fdfd-4d25-9fd2-2a46de44a9e6/children
```
Comment (bhapas) [inline packages/keycloak/data_stream/log/_dev/test/pipeline/test-log.log:22]: Can you add this after the multi line log just to make sure the pipeline picks it up correctly after the multiline log is done parsing.

**Review by efd6 — APPROVED** [review]
LGTM after @bhapas suggestion applied.

File: packages/keycloak/data_stream/log/_dev/test/pipeline/test-log.log (hunk 2 above)
Comment (chemamartinez) [reply packages/keycloak/data_stream/log/_dev/test/pipeline/test-log.log]: Sure!

**Review by bhapas — APPROVED** [review]

**Pull request merged**